mockkube/src/main/java/io/strimzi/test/mockkube/PredicatedWatcher.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.test.mockkube;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.client.Watcher;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.function.Predicate;
+
+/**
+ * A Watcher guarded by a Predicate; {@link #maybeFire(HasMetadata, Watcher.Action)}
+ * will call the watcher only if the predicate matches.
+ * This is used to support watches on resources matching selectors, for example.
+ * @param <CM> The resource type of the Watcher
+ */
+class PredicatedWatcher<CM extends HasMetadata> {
```

Comment:
"Predicated" sounds strange to me, maybe it's a problem of not being native speaker ... what does actually mean related to the implementation? That it's a watcher with a predicate? :-)

Reply:
Basically yes, the Watcher is only fired if it matches the predicate. Hence the Watcher is predicated on _whatever the predicate matches_.

See https://dictionary.cambridge.org/dictionary/english/predicated for a definition, though it seems not to be in a few online dictionaries. But I promise it really is a word and not something I've made up. I guess you're approval was not predicated on it being a real word though.